src/vm.rs

```diff
@@ -90,8 +91,18 @@ impl<VCpuType: VirtualCPU> UhyveVm<VCpuType> {
 	pub fn new(kernel_path: PathBuf, params: Params) -> HypervisorResult<UhyveVm<VCpuType>> {
 		let memory_size = params.memory_size.get();
 
+		// TODO: Move functionality to load_kernel. We don't know whether the binaries are relocatable yet.
+		// TODO: Use random address instead of arch::RAM_START here.
 		#[cfg(target_os = "linux")]
+		#[cfg(target_arch = "x86_64")]
 		let mem = MmapMemory::new(0, memory_size, arch::RAM_START, params.thp, params.ksm);
+
+		// TODO: guest_address is only taken into account on Linux platforms.
+		// TODO: Before changing this, fix init_guest_mem in `src/arch/aarch64/mod.rs`
+		#[cfg(target_os = "linux")]
+		#[cfg(not(target_arch = "x86_64"))]
```

Comment:
TODO: Remove this #[cfg... mess and replace it with a feature as soon as https://github.com/hermit-os/uhyve/pull/725/files#r1672908739 is answered.